src/Features/Core/Portable/Diagnostics/EngineV1/DiagnosticIncrementalAnalyzer_GetDiagnostics.cs

```diff
@@ -428,31 +453,25 @@ protected override Task AppendProjectStateDiagnosticsAsync(Project project, Docu
 
             private async Task<CompilationWithAnalyzers> GetCompilationWithAnalyzersAsync(Project project, bool concurrentAnalysis, bool reportSuppressedDiagnostics, CancellationToken cancellationToken)
             {
-                CompilationWithAnalyzers compilationWithAnalyzers;
-                lock (_compilationWithAnalyzersMap)
-                {
-                    if (_compilationWithAnalyzersMap.TryGetValue(project.Id, out compilationWithAnalyzers))
-                    {
-                        return compilationWithAnalyzers;
-                    }
-                }
-
-                if (project.SupportsCompilation)
+                if (_compilationWithAnalyzers == null && project.SupportsCompilation)
```

Comment:
This will break scenario where latest diagnostic getter is created for entire solution diagnostics. More specifically, I had introduced the map for FixAll occurrences in solution code path where we invoke compute diagnostics in parallel.